src/gpl/src/initialPlace.cpp

```diff
@@ -41,6 +43,40 @@
 #include "graphics.h"
 
 #include "utl/Logger.h"
+#include <cuda.h>
+#include <cuda_runtime.h>
+#include "cusparse.h"
+#include "cusolverDn.h"
+#include "cusolverSp.h"
+
+#define CUDA_ERROR(value) do { \
+   cudaError_t m_cudaStat = value; \
+   if(m_cudaStat != cudaSuccess){ \
+       fprintf(stderr, "Error %s at line %d in file %s\n", cudaGetErrorString(m_cudaStat), __LINE__, __FILE__); \
+       exit(-1); \
```

Comment:
Just to check we're following the expected convention: does the C++ code in this part of the code base generally exit on failures that we can't recover from? Are particular status codes expected? (e.g. how did we choose -5 below vs -1 here)?

Reply:
If the status is not "Success", the code will exit. But the number I picked here is random. I have changed them all to "-1" so that people know it is just a general failure.

Reply:
Do not use exit with any code - please see my comment about the logger.